portal-impl/test/unit/com/liferay/portal/zip/ZipWriterImplTest.java

```diff
@@ -305,6 +305,29 @@ public void testConstructor() {
 		file2.delete();
 	}
 
+	@Test
+	public void testConstructorWithFilePathContainSpecialCharacter()
+		throws IOException {
+
+		Path tempZipFilePath = Files.createTempFile(
+			Paths.get(SystemProperties.get(SystemProperties.TMP_DIR)), null,
+			"file path with space.zip");
+
+		Files.delete(tempZipFilePath);
```

Comment:
@JanisZhang I can see this is in the `setUp()` method, but it's still pretty weird. If the deleting file (clean-up-your-own-test-leftovers) logic is correctly implemented, you shouldn't need this. When you add try-finally in this test case, please remove this one.

Reply:
Hi @dantewang, force pushed. 
For this one,  I found that after `Files.createTempFile()`, it will create a binary (a text) file in the temp folder. 
after `ZipWriter zipWriter = new ZipWriterImpl(zipFile);` a zip will be created. So the `file.delete();` doesn't delete the binary file. 
```
public ZipWriterImpl(File file) {
		_file = file.getAbsoluteFile();

		URI fileURI = _file.toURI();

		_uri = URI.create("jar:file:" + fileURI.getRawPath());

		try (FileSystem fileSystem = FileSystems.newFileSystem(
				_uri, Collections.singletonMap("create", "true"))) {
		}
		catch (IOException ioException) {
			throw new UncheckedIOException(ioException);
		}
	}
```

Reply:
@JanisZhang Let's try this:

1. Create a temp folder `Path` using `Files` API.
2. Create the zip file object using `File zipFile = new File(tempFolderPath.toFile, "file with space.zip")`
3. Remove the file and the folder in finally.

After this I would like to find a time to apply this pattern to other test cases in `ZipWriterImplTest`. Creating a temp file and deleting it immediately is very strange.

Reply:
@dantewang sure.Thanks!